akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentBehaviors.scala

```diff
@@ -175,7 +168,35 @@ class PersistentBehavior[Command, Event, State](
    */
   def snapshotEvery(numberOfEvents: Long): PersistentBehavior[Command, Event, State] = {
     require(numberOfEvents > 0, s"numberOfEvents should be positive: Was $numberOfEvents")
-    copy(snapshotOn = (_, _, seqNr) ⇒ seqNr % numberOfEvents == 0)
+    copy(snapshotWhen = (_, _, seqNr) ⇒ seqNr % numberOfEvents == 0)
+  }
+
+  /**
+   * Change the journal plugin id that this actor should use.
+   */
+  def withPersistencePluginId(id: String): PersistentBehavior[Command, Event, State] = {
+    require(id != null, "persistence plugin id must not be null; use empty string for 'default' journal")
+    copy(journalPluginId = id)
+  }
+
+  /**
+   * Change the snapshot store plugin id that this actor should use.
+   */
+  def withSnapshotPluginId(id: String): PersistentBehavior[Command, Event, State] = {
+    require(id != null, "snapshot plugin id must not be null; use empty string for 'default' snapshot store")
+    copy(snapshotPluginId = id)
+  }
+
+  /**
+   * Changes the snapshot selection criteria used by this behavior.
+   * By default the most recent snapshot is used, and the remaining state updates are recovered by replaying events
+   * from the sequence number up until which the snapshot reached.
+   *
+   * You may configure the behavior to skip recovering snapshots completely, in which case the recovery will be
+   * performed by replaying all events -- which may take a long time.
+   */
+  def withSnapshotSelectionCriteria(selection: SnapshotSelectionCriteria): PersistentBehavior[Command, Event, State] = {
```

Comment:
Should we create typed versions of the API classes like `SnapshotSelectionCriteria` and `Recovery`?